flocker/node/agents/cinder.py

```diff
@@ -380,6 +380,31 @@ def _extract_nova_server_addresses(addresses):
     return all_addresses
 
 
+def _get_compute_id(local_ips, id_to_node_ips):
+    """
+    Compute the instance ID of the local machine.
+
+    Expectation is that our local IPs intersect with one (only) of the
+    remote nodes' sets of IPs.
+
+    :param set local_ips: The local machine's IPs.
+    :param id_to_node_ips: Mapping from instance IDs to sets of IPs, as
+        reported by OpenStack.
+
+    :return: Instance ID of local machine.
+    """
+    matching_instances = []
+    for server_id, api_addresses in id_to_node_ips.items():
```

Comment:
given the conversation on slack, its unclear to me how vital the need to do the metadata validations first would be. The only case I can think of is multiple virtual networks in the same openstack cluster that may report overlapping layer 2 addresses.